app/plugin_api.go

```diff
@@ -35,29 +35,23 @@ func NewPluginAPI(a *App, manifest *model.Manifest) *PluginAPI {
 }
 
 func (api *PluginAPI) applyDefaultConfigValues(pluginConfig map[string]interface{}) {
-	initialConfig := make(map[string]interface{})
 
-	if api.manifest.SettingsSchema != nil {
-		for _, settings := range api.manifest.SettingsSchema.Settings {
-			initialConfig[strings.ToLower(settings.Key)] = settings.Default
-		}
-	}
-	// Checking if plugin config was specified by user,
-	// if not we just apply default values from the initial config settings and return
-	if len(pluginConfig) == 0 {
-		for key, value := range initialConfig {
-			pluginConfig[key] = value
-		}
+	if api.manifest.SettingsSchema == nil {
 		return
 	}
-	// If plugin config was specified by user we check the values from plugin config and if there is not value for
-	// the key, we will apply default value from the initial plugin settings
-	for configKey, configValue := range initialConfig {
-		if pluginValue, exist := pluginConfig[configKey]; exist && pluginValue == "" {
-			pluginConfig[configKey] = configValue
+
+	if len(pluginConfig) != 0 {
```

Comment:
Perhaps I'm misunderstanding the loops, but I think we can simplify this even further and combine them together:
```go
for _, setting := range api.manifest.SettingsSchema.Settings {
    key = strings.ToLower(setting.Key)
    if existingValue := pluginConfig[key]; existingValue == "" {
        pluginConfig[key] = setting.Default
    }
}
```

The idea is that we need only iterate over the default values, and we need only apply them if the existing value is "" (or missing, which has a `nil` value of ""). No need to check if there are items in the list or not.

Reply:
@lieut-data Thank you for clarification. I was not add those changes in previous commit due to misunderstanding.

Reply:
Thanks, @igomonov88, and sorry for adding more confusion, but this block replaces /all/ the other code in this method. It's no longer necessary to test `len(pluginConfig) !=0` nor iterate over the `SettingsSchema` afterwards.

Can you simplify accordingly?